tests/everest/test_everserver.py

```diff
@@ -271,3 +273,34 @@ def test_everserver_status_max_batch_num(
         filter_out_gradient=False, batches=None
     )
     assert {data.batch for data in snapshot.simulation_data} == {0}
+
+
+def test_event_serialization(
+    copy_math_func_test_data_to_tmp,
+    evaluator_server_config_generator,
+):
+    config = EverestConfig.load_file("config_minimal.yml")
+
+    def check_status_round_tripping(status):
+        round_trip_status = json.loads(json.dumps(status))
+        assert round_trip_status == status
+
+    run_model = EverestRunModel.create(
+        config,
+        simulation_callback=check_status_round_tripping,
+    )
+
+    send_snapshot_event = run_model.send_snapshot_event
+
+    def check_event_serialization_round_trip(*args, **_):
+        event, _ = args
+        event_json = event_to_json(event)
+        round_trip_event = event_from_json(str(event_json))
+        assert event == round_trip_event
```

Comment:
In this case, if we get a `FullSnapshotEvent` or a `SnapshotUpdateEvent`, are we able to differentiate them?

Reply:
Good point